contributors/design-proposals/core-metrics-pipeline.md

```diff
@@ -0,0 +1,295 @@
+
+The target "Users" of this set of metrics are kubernetes components.  This set of metrics itself is not designed to be user-facing, but is designed to be general enough to support user-facing components.
+
+### Non Goals
+Everything covered in the [Monitoring Architecture](https://github.com/kubernetes/kubernetes/blob/master/docs/design/monitoring_architecture.md) design doc will not be covered in this proposal.  This includes the third party metrics pipeline, and the methods by which the metrics found in this proposal are provided to other kubernetes components.
+
+Integration with CRI will not be covered in this proposal.  In future proposals, integrating with CRI may provide a better abstraction of information required by the core metrics pipeline to collect metrics.
+
+## Design
+This design covers only the Core Metrics Pipeline.
+
+High level requirements for the design are as follows:
+ - Do not break existing users.  We should continue to provide the full summary API as an optional add-on.  Once the monitoring pipeline is completed, the summary API will be provided by the monitoring pipeline, possibly through a stand-alone version of cAdvisor.
+ - The kubelet collects the minimum possible number of metrics for complete portable kubernetes functionalities.
+ - Metrics can be fetched "On Demand", giving the kubelet more up-to-date stats.
+
+More details on how these high level goals will be achieved can be found in the Implementation Plan.
+
+This Core Metrics API will be versioned to account for version-skew between kubernetes components.
+
+This proposal purposefully omits many metrics that may eventually become core metrics.  This is by design.  Once metrics are needed to support resource feasibility checking or node resource management, they can be added to the core metrics API.
+
+### Metric Requirements
+The core metrics api is designed to provide metrics for two use-cases within kubernetes:
+ - Resource Feasibility Checking
+ - Node Resource Management
+
+Many kubernetes system components currently support these features.  Many more components that support these features are in development.
+The following is not meant to be an exhaustive list, but gives the current set of use cases for these metrics.
+
+Metrics requirements for resource feasibility checking and node resource management, based on kubernetes component needs, are as follows:
+ - Kubelet
+  - Node-level capacity and availability metrics for Disk, Memory, and CPU
+  - Pod-level usage metrics for Disk and Memory
+ - Scheduler (Possibly through [Resource Metrics API](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/resource-metrics-api.md))
+  - Node-level capacity and availability metrics for Disk, CPU, and Memory
+  - Pod-level usage metrics for Disk, CPU, and Memory
+  - Container-level usage metrics for Disk, CPU, and Memory
```

Comment:
Why does the scheduler need container-level usage? Or does this just mean as distinct from pod overhead?